tests/integration/conftest.py

```diff
@@ -48,6 +49,127 @@
 WORKER_COUNTER_KEY = "worker_counter"
 
 
+def _clear_test_name(name: str) -> str:
+    # Based on:
+    # https://github.com/pytest-dev/pytest/blob/238b25ffa9d4acbc7072ac3dd6d8240765643aed/src/_pytest/tmpdir.py#L189-L194
+    # The purpose is to return a similar name to make finding matching logs easier
+    return re.sub(r"[\W]", "_", name)[:30]
+
+
+@dataclass(frozen=True)
+class RegistryEndpoint:
+    protocol: str
+    host: str
+    port: int
+
+    def to_url(self) -> str:
+        return f"{self.protocol}://{self.host}:{self.port}"
+
+
+@dataclass(frozen=True)
+class RegistryDescription:
+    endpoint: RegistryEndpoint
+    schemas_topic: str
+
+
+@asynccontextmanager
+async def start_schema_registry_cluster(
```

Comment:
This is the main fix. It will start a cluster with a new schemas topic and a new group id, which prevents most of the issues with flakiness. I also tried to be careful to now leak any servers behind, as that would cause subsequent runs to fail too